src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs

```diff
@@ -2,55 +2,193 @@
+using System.Collections.Generic;
 using System.Threading;
 using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Logging;
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Hosting;
 using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
 
 namespace Microsoft.Azure.WebJobs.Script.WebHost
 {
-    public class WebJobsScriptHostService : IHostedService, IDisposable
+    public class WebJobsScriptHostService : IHostedService, IScriptHostManager, IDisposable
     {
-        private readonly WebScriptHostManager _scriptHostManager;
-        private readonly CancellationTokenSource _cancellationTokenSource;
+        private readonly IServiceProvider _rootServiceProvider;
+        private readonly IServiceScopeFactory _rootScopeFactory;
+        private readonly IOptions<ScriptApplicationHostOptions> _webHostOptions;
         private readonly ILogger _logger;
+        private CancellationTokenSource _startupLoopTokenSource;
         private bool _disposed = false;
-        private Task _hostTask;
+        private IHost _host;
 
-        public WebJobsScriptHostService(WebScriptHostManager scriptHostManager, ILoggerFactory loggerFactory)
+        public WebJobsScriptHostService(IOptions<ScriptApplicationHostOptions> webHostOptions, IServiceProvider rootServiceProvider, IServiceScopeFactory rootScopeFactory, ILoggerFactory loggerFactory)
         {
-            _scriptHostManager = scriptHostManager ?? throw new ArgumentException($@"Unable to locate the {nameof(WebScriptHostManager)} service. " +
-                    $"Please add all the required services by calling '{nameof(IServiceCollection)}.{nameof(WebJobsServiceCollectionExtensions.AddWebJobsScriptHost)}' " +
-                    $"inside the call to 'ConfigureServices' in the application startup code");
+            if (loggerFactory == null)
+            {
+                throw new ArgumentNullException(nameof(loggerFactory));
+            }
+
+            _rootServiceProvider = rootServiceProvider ?? throw new ArgumentNullException(nameof(rootServiceProvider));
+            _rootScopeFactory = rootScopeFactory ?? throw new ArgumentNullException(nameof(rootScopeFactory));
+            _webHostOptions = webHostOptions ?? throw new ArgumentNullException(nameof(webHostOptions));
 
-            _cancellationTokenSource = new CancellationTokenSource();
-            _hostTask = Task.CompletedTask;
             _logger = loggerFactory.CreateLogger(ScriptConstants.LogCategoryHostGeneral);
+
+            State = ScriptHostState.Default;
+        }
+
+        public IServiceProvider Services => _host?.Services;
+
+        public ScriptHostState State { get; private set; }
+
+        public Exception LastError { get; private set; }
+
+        public async Task StartAsync(CancellationToken cancellationToken)
+        {
+            _startupLoopTokenSource = new CancellationTokenSource();
+            var startupLoopToken = _startupLoopTokenSource.Token;
```

Comment:
So ScriptHostManager went away and this is its replacement. Where did some of the logic that used to exist go - e.g. HostHealthMonitor/Shutdown